LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift

```diff
@@ -11,18 +11,19 @@ import SwiftUI
 import LoopKit
 
 public struct InsulinModelSelection: View {
+    @EnvironmentObject private var displayGlucoseUnitObservable: DisplayGlucoseUnitObservable
     @Environment(\.appName) private var appName   
-    @Environment(\.dismissAction) var dismiss
-    @Environment(\.authenticate) var authenticate
-
-    let initialValue: InsulinModelSettings
-    @State var value: InsulinModelSettings
-    let insulinSensitivitySchedule: InsulinSensitivitySchedule
-    let glucoseUnit: HKUnit
-    let supportedModelSettings: SupportedInsulinModelSettings
-    let mode: SettingsPresentationMode
-    let save: (_ insulinModelSettings: InsulinModelSettings) -> Void
-    let chartManager: ChartsManager
+    @Environment(\.dismissAction) private var dismiss
+    @Environment(\.authenticate) private var authenticate
+
+    @State private var value: InsulinModelSettings
+    @State private var chartManager: ChartsManager
```

Comment:
[nit] Was making `chartManager` a `@State` intentional?

Reply:
Yes, since a new chartManager was being creating with each refresh, the reference to one controlling the displayed chart unit was lost and thus it could not be used to triggered a chart update.